lib/before-prepare.js

```diff
@@ -11,5 +11,6 @@ module.exports = function ($logger, $projectData, $options, hookArgs) {
 	}
 
 	var release = $options.release || appFilesUpdaterOptions.release;
-	return compiler.runTypeScriptCompiler($logger, $projectData.projectDir, { release });
+	var syncAllFiles = $options && $options.syncAllFiles;
```

Comment:
this will not work in case the hook is executed from Sidekick (i.e. when CLI is used as a library). In this case the `$options` values will be the default ones as there's no command line args to populate them. Instead of using `$options`, you'll have to modify the [preparePlatformCore](https://github.com/NativeScript/nativescript-cli/blob/master/lib/services/platform-service.ts#L273) method to receive this value as an argument. For example, we can add `skipModulesNativeCheck` (as there's already such value in the `preparePlatform` method) in the `appFilesUpdaterOptions` object and you can access it here via:
```TypeScript
const skipModulesNativeCheck = hookArgs && hookArgs.appFilesUpdaterOptions && hookArgs.appFilesUpdaterOptions.skipModulesNativeCheck;
const syncAllFiles = $options && $options.syncAllFiles || !skipModulesNativeCheck;
```